models/_sector/dex/bot_trades/solana/platforms/consortium_key_solana_bot_trades.sql

```diff
@@ -0,0 +1,118 @@
+{{
+    config(
+        alias="bot_trades",
+        schema="consortium_key_solana",
+        partition_by=["block_month"],
+        materialized="incremental",
+        file_format="delta",
+        incremental_strategy="merge",
+        unique_key=[
+            "blockchain",
+            "tx_id",
+            "tx_index",
+            "outer_instruction_index",
+            "inner_instruction_index",
+        ],
+    )
+}}
+
+{% set project_start_date = "2024-03-01" %}
+{% set fee_receiver_1 = "6qgwjhV2RQxcPffRdtQBTTEezRykQKXqhcDyv1z3r9tq" %}
+{% set wsol_token = "So11111111111111111111111111111111111111112" %}
+
+with
+    users as (
+        select distinct transactions.signer as user
+        from {{ source("solana", "account_activity") }} as activity
+        join {{ source("solana", "transactions") }} as transactions on tx_id = id
+        where
+            activity.block_time >= timestamp '{{project_start_date}}'  -- FirstTradeTimestamp
+            and transactions.block_time >= timestamp '{{project_start_date}}'  -- FirstTradeTimestamp
+            and tx_success
+            and address = '{{fee_receiver_1}}'  -- SolanaFeeWallet
+            and balance_change > 0  -- SOL fee payments
+    ),
```

Comment:
is this model unique and requires this CTE, or can we avoid it? i notice most others in this spell lineage do this:
https://github.com/duneanalytics/spellbook/blob/main/models/_sector/dex/bot_trades/solana/platforms/bonkbot_solana_bot_trades.sql#L95-L110
which seems to be more efficient runtime. 
notice in CI runs here, the incremental step takes 10+ minutes, which appears to be due to this CTE which takes ~6 minutes to return results on it's own

Reply:
since we need to fetch all users in order to fetch their trades, i guess we can not get rid of this unfortunately @jeff-dude

Reply:
can we at least look to be consistent between the two? one filters incrementally on account activity, while the other does not. if possible, please add incremental filter to transactions too

Reply:
i guess we should remove the incremental filter from the other spellbook. correct me if i'm wrong but when we do incremental filters here, we would probably miss historical users?

Reply:
most likely, yes. is this the use case?
- new trades in dex solana could have users from months ago, so if we filter incrementally on users, it would find no match.
- if that isn't the case, then we could be safe to assume new trades find new users

do we know how often new users are output in these CTEs? would it make sense to make their own spell upstream? we could incrementally build that spell to add new users, then call the spell in these ones for much faster runtime

Reply:
sorry i don't understand what you're trying to say here. the methodology in the CTE is the only way to find solana users, since CK does not pay fees in the trade txs, but waits for a certain threshold until it sends them to the fee address

if the problem is dex.bot_trades, then we could just not include it there, the important thing here is that the data goes live, so the CK dashboard will have solana metrics again

Reply:
the concern is [here](https://github.com/duneanalytics/spellbook/actions/runs/9156694489/job/25171548525?pr=5880#step:14:19). that's a long runtime for incremental runs. you can see in [other PR](https://github.com/duneanalytics/spellbook/actions/runs/9157874643/job/25175092596?pr=5881#step:14:21), the two models which don't use this CTE logic run in ~60s during incremental step

i'd like to see that improved potentially by removing the CTE in the slow ones and making the CTE it's own spell. then replace CTE with that spell in join conditions. can we try that?

Reply:
sure we can! would it be possible to get this merged before? and then i'll submit a new PR with the additional spellbook + changes in this one?

Reply:
that works for me. i'll get this in & then we can work on the enhancements.